### PR TITLE
[289510] Implement OpenApiAssertHelper

### PR DESCRIPTION
**Pull request by aYukiSekiguchi**
https://acsmine.tok.access-company.com/redmine/issues/289510

### CODE REVIEW

File: lib/test/openapi_assert_helper.ex [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+        ref_targets =
+          schema["allOf"]
+          # e.x. item %{"$ref" => [:root, "components", "schemas", "AdminUser"]},
+          |> Enum.filter(fn x -> Map.has_key?(x, "$ref") end)
+          |> Enum.map(fn x ->
+            [:root, "components" | rest_path] = x["$ref"]
+            rest_path
+          end)
+
+          # e.x. item ["components", "schemas", "AdminUser"]
```
Comment (Double-oxygeN) [inline lib/test/openapi_assert_helper.ex:634]: `"components"` is already dropped at above.

```suggestion
          # e.g. item ["schemas", "AdminUser"]
```

File: lib/test/openapi_assert_helper.ex [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+            %{status: _status} = res :: v[Response.t()]
+          ) ::
+            :ok do
+      if res.status == 500 do
+        IO.puts(res.body)
+      end
+
+      operation_id = asserter.api["operationId"]
+      expected_status = success_response_status(asserter)
+      assert {res.status, operation_id, res.body} == {expected_status, operation_id, res.body}
```
[the remark this answers is not in the document]
Comment (aYukiSekiguchi) [reply lib/test/openapi_assert_helper.ex:415]: This makes easy to understand when assertion is failed. Your suggestion just show `200 != 400`. It's hard to understand what REST API was failed, or it shows nothing about what detaild error code was written in the body. If we add `operation_id` and `res.body` to the `assert`, it automatically shows `operation_id` and `res.body` when `res.status != expected_status`.

[the remark this answers is not in the document]
Comment (Double-oxygeN) [reply lib/test/openapi_assert_helper.ex:415]: It looks quite nice. Thank you for helping my understanding.

File: lib/test/openapi_assert_helper.ex [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+    defunp pick_success_response(asserter :: SchemaAsserter.t()) ::
+             v[{non_neg_integer, map | nil}] do
+      responses = asserter.api["responses"]
+
+      success_key =
+        responses
+        |> Map.keys()
+        |> Enum.find(fn key ->
+          # credo:disable-for-next-line Credo.Check.Refactor.AppendSingleItem
+          key in (200..207 |> Enum.map(&to_string/1))
```
[the remark this answers is not in the document]
Comment (aYukiSekiguchi) [reply lib/test/openapi_assert_helper.ex]: Thank you.
No. Removed at https://github.com/access-company/antikythera/pull/273/commits/f56774836796e16203372538ee7bc9fccfa6848d

File: lib/test/openapi_assert_helper.ex [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+             asserter :: SchemaAsserter.t(),
+             body :: v[map],
+             status :: v[non_neg_integer]
+           ) ::
+             v[:ok | {:error, map | :status_not_found}] do
+      api_error =
+        asserter.api["responses"]
+        |> Enum.find(fn {key, _} ->
+          # credo:disable-for-next-line Credo.Check.Refactor.AppendSingleItem
+          api_status = String.to_integer(key)
```
[the remark this answers is not in the document]
Comment (aYukiSekiguchi) [reply lib/test/openapi_assert_helper.ex]: Removed at https://github.com/access-company/antikythera/pull/273/commits/f56774836796e16203372538ee7bc9fccfa6848d

File: lib/test/openapi_assert_helper.ex (hunk 1 above)
Comment (aYukiSekiguchi) [reply lib/test/openapi_assert_helper.ex]: Good catch!
Fixed at https://github.com/access-company/antikythera/pull/273/commits/aa0399d8e008cee7e8a405489b358e605e9d381a

File: lib/test/openapi_assert_helper.ex [hunk 5]
```diff
@@ -0,0 +1,693 @@
...
+      alias Antikythera.Httpc.Response
+      alias Antikythera.Test.OpenApiAssertHelper.{Normalizer, SchemaAsserter}
+      import ExUnit.Assertions
+
+      @assert_option_keys [:ignore_req_fields]
+
+      @doc """
+      Send a GET request which should be success.
+
+      - `api_schema`: Schema definition from `__MODULE__.find_api/1`
```
[the remark this answers is not in the document]
Comment (Double-oxygeN) [reply lib/test/openapi_assert_helper.ex]: Would you apply the same change for other docs?

[the remark this answers is not in the document]
Comment (aYukiSekiguchi) [reply lib/test/openapi_assert_helper.ex]: Sorry. Fixed at https://github.com/access-company/antikythera/pull/273/commits/8fed795d771f3636b996d908ae3a8536b15b88e3

**Review by Double-oxygeN — APPROVED** [review]
LGTM

**Pull request merged**